httpretty/core.py

```diff
@@ -725,6 +727,15 @@ def matches(self, info):
             return self.regex.search(info.full_url(
                 use_querystring=self._match_querystring))
 
+    def check_expected_data(self, request):
+        if self.expected_data is not None:
+            body_dict = dict(parse_qsl(request.body))
+            if body_dict != self.expected_data:
+                raise ValueError("Body Post didn't match, expected %s, got %s" % (
```

Comment:
this should raise an `AssertionError` instead, that way the test will fail as *failure* rather than *error*